pkg/relayer/config/relayminer_configs_reader.go

```diff
@@ -4,98 +4,264 @@ import (
-}
-
-// RelayMinerConfig is the structure describing the RelayMiner config
-type RelayMinerConfig struct {
-	QueryNodeUrl            *url.URL
-	NetworkNodeUrl          *url.URL
-	PocketNodeWebsocketUrl  string
-	SigningKeyName          string
-	ProxiedServiceEndpoints map[string]*url.URL
-	SmtStorePath            string
-}
-
 // ParseRelayMinerConfigs parses the relay miner config file into a RelayMinerConfig
 func ParseRelayMinerConfigs(configContent []byte) (*RelayMinerConfig, error) {
-	var yamlRelayMinerConfig YAMLRelayMinerConfig
+	var (
+		yamlRelayMinerConfig YAMLRelayMinerConfig
+		err                  error
+	)
+
+	// The config file should not be empty
+	if len(configContent) == 0 {
+		return nil, ErrRelayMinerConfigEmpty
+	}
 
 	// Unmarshal the stake config file into a yamlAppGateConfig
 	if err := yaml.Unmarshal(configContent, &yamlRelayMinerConfig); err != nil {
-		return nil, ErrRelayMinerConfigUnmarshalYAML.Wrapf("%s", err)
+		return nil, ErrRelayMinerConfigUnmarshalYAML.Wrap(err.Error())
 	}
 
-	// Check that the query node URL is provided
-	if yamlRelayMinerConfig.QueryNodeUrl == "" {
-		return nil, ErrRelayMinerConfigInvalidQueryNodeUrl.Wrapf("query node url is required")
+	// Top level section
+	// SigningKeyName is required
+	if yamlRelayMinerConfig.SigningKeyName == "" {
+		return nil, ErrRelayMinerConfigInvalidSigningKeyName
 	}
 
-	// Parse the query node URL
-	queryNodeUrl, err := url.Parse(yamlRelayMinerConfig.QueryNodeUrl)
-	if err != nil {
-		return nil, ErrRelayMinerConfigInvalidQueryNodeUrl.Wrapf("%s", err)
+	// SmtStorePath is required
+	if yamlRelayMinerConfig.SmtStorePath == "" {
+		return nil, ErrRelayMinerConfigInvalidSmtStorePath
 	}
 
-	// Check that the network node URL is provided
-	if yamlRelayMinerConfig.NetworkNodeUrl == "" {
-		return nil, ErrRelayMinerConfigInvalidNetworkNodeUrl.Wrapf("network node url is required")
+	// Pocket node urls section
+	relayMinerPocketConfig := &RelayMinerPocketConfig{}
+	pocket := yamlRelayMinerConfig.Pocket
+
+	if pocket.TxNodeGRPCUrl == "" {
+		return nil, ErrRelayMinerConfigInvalidNodeUrl.Wrap("tx node grpc url is required")
 	}
 
-	// Parse the network node URL
-	networkNodeUrl, err := url.Parse(yamlRelayMinerConfig.NetworkNodeUrl)
+	// Check if the pocket node grpc url is a valid URL
+	relayMinerPocketConfig.TxNodeGRPCUrl, err = url.Parse(pocket.TxNodeGRPCUrl)
 	if err != nil {
-		return nil, ErrRelayMinerConfigInvalidNetworkNodeUrl.Wrapf("%s", err)
+		return nil, ErrRelayMinerConfigInvalidNodeUrl.Wrapf(
+			"invalid tx node grpc url %s",
+			err.Error(),
+		)
 	}
 
-	// Parse the websocket URL of the Pocket Node to connect to for subscribing to on-chain events.
-	pocketNodeWebsocketUrl := sdk.HostToWebsocketURL(queryNodeUrl.Host)
+	// If the query node grpc url is empty, use the tx node grpc url
+	if pocket.QueryNodeGRPCUrl == "" {
+		relayMinerPocketConfig.QueryNodeGRPCUrl = relayMinerPocketConfig.TxNodeGRPCUrl
+	} else {
```

Comment:
Not sure what you want to achieve here.

We want to use `TxNodeGRPCUrl` which is already validated above., if `QueryNodeGRPCUrl` is not set.
Otherwise (`QueryNodeGRPCUrl` has its own value) we need to make sure it's a valid URL.

Reply:
I thought it was just an easier way to write the code (more readable) but feel free to ignore if that's not the case from your POV.